merino-cache/src/memory.rs

```diff
@@ -207,33 +218,51 @@ impl SuggestionProvider for Suggester {
                 }
             }
 
-            if LOCK_TABLE.is_locked(&key) {
-                // there's a fetch already in progress. Return empty for now.
-                return Ok(SuggestionResponse {
-                    cache_status: CacheStatus::Hit,
-                    cache_ttl: None,
-                    suggestions: Vec::new(),
-                });
+            if rv.is_none() {
+                if LOCK_TABLE.is_locked(&key) {
+                    // there's a fetch already in progress. Return empty for now.
+                    rv = Some(SuggestionResponse {
+                        cache_status: CacheStatus::Hit,
+                        cache_ttl: None,
+                        suggestions: Vec::new(),
+                    });
+                } else {
+                    // handle cache miss or stale cache
+                    let lock = LOCK_TABLE.add_lock(&key, self.default_lock_timeout);
+                    let mut response = self
+                        .inner
+                        .suggest(query)
+                        .await?
+                        // Todo, cache status should be a vec.
```

Comment:
Should this be turned into an issue on GH/Jira Ticket and have it referenced here?

Reply:
I'm going to just take the comment out for now; I think we should see how we end up using the caches before we invest more into this. The possibilities I can think of already are, which are not mutually exclusive:

1. We only ever use one cache per provider-chain, so we only need a single cache status
2. We often pair memory and redis (or another) cache, and want to report cache status for both
3. We often use Multi or IdMulti, which may want to report cache statuses from multiple providers independently
4. We want to report cache status in metrics or logs
5. We want to also include the full cache picture in the HTTP responses

Once we know more about what we want/need out of this area, lets revisit this.